From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Fri, 21 Aug 2026 15:25:00 +0200
Subject: [PATCH] Allow SciPy pytest to start on Emscripten

Emscripten has no fork/forkserver. Do not force that start method so
scipy.test() can collect. This patch does not skip any tests.

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 scipy/conftest.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/scipy/conftest.py b/scipy/conftest.py
index 2b82871..a1b2c3d 100644
--- a/scipy/conftest.py
+++ b/scipy/conftest.py
@@ -97,7 +97,9 @@ def pytest_configure(config):
             "iterations(n): run the given test function `n` times in each thread",
         )
 
-    if os.name == 'posix' and sys.version_info < (3, 14) and sys.platform != "cygwin":
+    # Emscripten has no fork/forkserver.
+    if (os.name == 'posix' and sys.version_info < (3, 14)
+            and sys.platform not in ("cygwin", "emscripten")):
         # On POSIX, Python 3.13 and older uses the 'fork' context by
         # default. Calling fork() from multiple threads leads to
         # deadlocks. This has been changed in 3.14 to 'forkserver'.
-- 
2.55.0
